common/current_preset.h

```diff
@@ -271,10 +271,30 @@ class CurrentPreset {
 
   void SetPreset(int preset) {
     Clear();
+    #ifdef SAVED_PRESET
+    SavePreset(preset);
+    #endif
     LOCK_SD(true);
     if (!Load(preset)) Set(preset);
     LOCK_SD(false);
   }
+
+  bool SavePreset(int preset) {
```

Comment:
This doesn't actually save the preset, but it saves what preset we are currently on.
Also, it saves the volume.
I don't think we should take a number as argument here, as we can get the preset number from the current preset.

Reply:
Maybe it would be better to add this to presets.ini?

Reply:
I think it's better to keep it separate because it's doing a different job.  Save State is a bookmark to where the saber was when it was powered off last.